tests/ui/panic_unimplemented.stderr

```diff
@@ -1,28 +1,37 @@
 error: you probably are missing some parameter in your format string
- --> $DIR/panic.rs:8:16
+ --> $DIR/panic_unimplemented.rs:8:16
   |
 8 |         panic!("{}");
   |                ^^^^
   |
   = note: `-D panic-params` implied by `-D warnings`
 
 error: you probably are missing some parameter in your format string
-  --> $DIR/panic.rs:10:16
+  --> $DIR/panic_unimplemented.rs:10:16
    |
 10 |         panic!("{:?}");
    |                ^^^^^^
 
 error: you probably are missing some parameter in your format string
-  --> $DIR/panic.rs:12:23
+  --> $DIR/panic_unimplemented.rs:12:23
    |
 12 |         assert!(true, "here be missing values: {}");
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 error: you probably are missing some parameter in your format string
-  --> $DIR/panic.rs:15:12
+  --> $DIR/panic_unimplemented.rs:15:12
    |
 15 |     panic!("{{{this}}}");
    |            ^^^^^^^^^^^^
 
-error: aborting due to 4 previous errors
+error: `unimplemented` should not be present in production code
+  --> $DIR/panic_unimplemented.rs:57:5
+   |
+57 |     unimplemented!();
+   |     ^^^^^^^^^^^^^^^^^
+   |
+   = note: `-D unimplemented` implied by `-D warnings`
+   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

Comment:
can you try getting rid of the "originates in a macro" note by using a span that isn't inside the expansion? Not sure which one that would be

Reply:
I found a way to avoid the "originates in a macro", by changing the first part of the check to
```
if let ExprBlock(ref block, _) = expr.node {
  for stmt in &block.stmts {
    if_chain! {
      if let StmtSemi(ref exp, _) = stmt.node;
         ...
```

When I then run `is_expn_of` on `block.span` it finds the unimplemented call, but it marks too big of an area:
```
error: `unimplemented` should not be present in production code
  --> tests/ui/panic_unimplemented.rs:56:20
   |
56 |   fn unimplemented() {
   |  ____________________^
57 | |     let a = 2;
58 | |     unimplemented!();
59 | |     let b = a + 2;
60 | | }
   | |_^
   |
   = note: `-D unimplemented` implied by `-D warnings`
```

Going one step deeper, and checking on `stmt.span` ends up with the correct area marked but then I get the note about macro origin again.

Reply:
hmm... since we do know we're in the expansion context of `unimplemented!`, we can just go up one level of macro expansions. I have no idea how to do that though.

If you want to tackle it, have a look at the type that is returned when you call `span.ctxt()` It should have ways to get the parent context. Then just overwrite the context of the span with the parent context and report the lint there.

Otherwise just open an issue so we fix it later